### ISSUE TITLE
Installation of Anaconda on all yarn nodes...

### ISSUE THREAD

**Issue by kkalugerov** [jupyter-incubator/enterprise_gateway#348]
The PATH of the anaconda of all nodes, should it be the same? for example /home/user/anaconda3.0 or it can be different on every node :?

**Comment by akchinSTC** [comment]
I don't think the PATH to conda installations have to be consistent on all nodes....correct me if im wrong @kevin-bates Ideally they would be identical on all nodes to make managing configuration and package/library management easier, especially across larger clusters, but if you happen to have anaconda installed in a different directories... you should be able to make changes in each respective kernel.json on each node.

**Comment by lresende** [comment]
My understanding is that the requirement is to actually have a homogeneous environment regarding the version of Python in use and don't mix for example Python 2.x with Python 3.x or even Python 3.x with Python 3.y. Having said that, all our code is actually relying on the Python environment on the path, except for running on cluster mode which we then explicitly point to the environment related to the Spark/YARN distribution in use, e.g:

```
{
  "language": "python",
  "display_name": "Spark - Python (YARN Cluster Mode)",
  "process_proxy": {
    "class_name": "enterprise_gateway.services.processproxies.yarn.YarnClusterProcessProxy"
  },
  "env": {
    "SPARK_HOME": "/usr/hdp/current/spark2-client",
    "PYSPARK_PYTHON": "/opt/anaconda2/bin/python",
    "PYTHONPATH": "${HOME}/.local/lib/python2.7/site-packages:/usr/hdp/current/spark2-client/python:/usr/hdp/current/spark2-client/python/lib/py4j-0.10.4-src.zip",
    "SPARK_OPTS": "--master yarn --deploy-mode cluster --name ${KERNEL_ID:-ERROR__NO__KERNEL_ID} --conf spark.yarn.submit.waitAppCompletion=false --conf spark.yarn.appMasterEnv.PYTHONUSERBASE=/home/yarn/.local --conf spark.yarn.appMasterEnv.PYTHONPATH=${HOME}/.local/lib/python2.7/site-packages:/usr/hdp/current/spark2-client/python:/usr/hdp/current/spark2-client/python/lib/py4j-0.10.4-src.zip --conf spark.yarn.appMasterEnv.PATH=/opt/anaconda2/bin:$PATH",
    "LAUNCH_OPTS": ""
  },
  "argv": [
    "/usr/local/share/jupyter/kernels/spark_python_yarn_cluster/bin/run.sh",
    "{connection_file}",
    "--RemoteProcessProxy.response-address",
    "{response_address}",
    "--RemoteProcessProxy.port-range",
    "{port_range}",
    "--RemoteProcessProxy.spark-context-initialization-mode",
    "lazy"
  ]
}
```

In summary, divergentes locations for the Anaconda installation should work as long as it's configured in the path.

**Comment by kevin-bates** [comment]
Closing issue.  Please re-open if you feel the question was not adequately answered or you have a follow-on question.
